x/merkledb/db.go

```diff
@@ -234,53 +221,70 @@ func newDatabase(
 	}
 
 	// mark that the db has not yet been cleanly closed
-	err = trieDB.metadataDB.Put(cleanShutdownKey, didNotHaveCleanShutdown)
+	err = trieDB.underlyingDB.Put(cleanShutdownKey, didNotHaveCleanShutdown)
 	return trieDB, err
 }
 
 // Deletes every intermediate node and rebuilds them by re-adding every key/value.
 // TODO: make this more efficient by only clearing out the stale portions of the trie.
 func (db *merkleDB) rebuild(ctx context.Context) error {
 	db.root = newNode(nil, RootPath)
-	if err := db.nodeDB.Delete(rootKey); err != nil {
+	if err := db.intermediateNodesDB.Delete(RootPath); err != nil {
+		return err
+	}
+	if err := db.valueNodesDB.Delete(RootPath); err != nil {
```

Comment:
If we shut down mid way through calling this (and this actually deleted the value) - wouldn't this result in a permanent DB corruption? I guess I'm confused why we delete the root node if it is in the value db.